scripts/abi_check.py

```diff
@@ -120,16 +123,30 @@ def update_git_submodules(self, git_worktree_path, crypto_rev):
             raise Exception("git submodule update failed, aborting")
         if (os.path.exists(os.path.join(git_worktree_path, "crypto"))
                 and crypto_rev):
-            checkout_process = subprocess.Popen(
-                [self.git_command, "checkout", crypto_rev],
-                cwd=os.path.join(git_worktree_path, "crypto"),
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT
-            )
-            checkout_output, _ = checkout_process.communicate()
-            self.log.info(checkout_output.decode("utf-8"))
-            if checkout_process.returncode != 0:
-                raise Exception("git checkout failed, aborting")
+            if crypto_repo:
+                shutil.rmtree(os.path.join(git_worktree_path, "crypto"))
+                clone_process = subprocess.Popen(
+                    [self.git_command, "clone", crypto_repo,
+                     "--branch", crypto_rev, "crypto"],
+                    cwd=git_worktree_path,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT
+                )
+                clone_output, _ = clone_process.communicate()
```

Comment:
It'd be much more efficient than doing a new clone if we added a new remote for the repo requested, fetched through there, and then did a checkout. `git remote add repo git@blahblah:blah/blah.git && git fetch repo && git checkout deadbeef123`. All hashes and branches already fetched wouldn't need to get fetched again; only what's different would be fetched.

Reply:
I'll try this and see what its like.

Reply:
How was it?

Reply:
This is much better, I've reworked it.